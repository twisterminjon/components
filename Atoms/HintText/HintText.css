.hinttext {
  font-size: 0.8rem;
  color: var(--misc__subtle__text_color);
  white-space: nowrap;
  margin-right: 10px;
  overflow: hidden;
}
