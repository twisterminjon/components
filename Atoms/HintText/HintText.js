import React, { Component } from 'react'
import PropTypes from 'prop-types'
import './HintText.css'

export default class HintText extends Component {
  static propTypes = {
    /** Text for Hint */
    hint: PropTypes.string,
    /** Align to right */
    rightAlign: PropTypes.bool,
  }
  static defaultProps = {
    hint: '',
    rightAlign: false,
  }

  render() {
    const { hint, rightAlign, ...rest } = this.props

    const rightMargin = rightAlign ? { marginLeft: 'auto' } : { marginLeft: 8 }

    return (
      <span className="hinttext" style={rightMargin} {...rest}>
        {hint}
      </span>
    )
  }
}
