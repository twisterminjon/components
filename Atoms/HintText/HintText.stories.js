import React from 'react'

import { storiesOf } from '@storybook/react'
import centered from '@storybook/addon-centered/react'
import { withKnobs, text } from '@storybook/addon-knobs'

import HintText from './HintText'

const stories = storiesOf('Provider/v1/Atoms', module)
stories.addDecorator(withKnobs)

stories.addDecorator(centered).add('HintText', () => {
  const hintLabel = 'hint'
  const hintDefault = 'Psst...I know your secrets'
  const hint = text(hintLabel, hintDefault)

  return <HintText hint={hint} />
})
