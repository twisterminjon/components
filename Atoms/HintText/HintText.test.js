import React from 'react'
import HintText from './HintText'

describe('HintText', () => {
  const hintValue = 'I am a hint'

  it('matches the snapshot', () => {
    const wrapper = window.shallow(<HintText hint={hintValue} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('contains the correct hint text', () => {
    const wrapper = window.shallow(<HintText hint={hintValue} />)
    expect(wrapper.text()).toEqual(hintValue)
  })
})
