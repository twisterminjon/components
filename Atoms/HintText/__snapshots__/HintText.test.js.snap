// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HintText matches the snapshot 1`] = `
<span
  className="hinttext"
  style={
    Object {
      "marginLeft": 8,
    }
  }
>
  I am a hint
</span>
`;
